src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderList.tsx

```diff
@@ -0,0 +1,198 @@
+import { COW_SWAP_NATIVE_ASSET_MARKER_ADDRESS } from '@shapeshiftoss/swapper/dist/swappers/CowSwapper/utils/constants'
+import { OrderClass, OrderStatus, SigningScheme } from '@shapeshiftoss/types/dist/cowSwap'
+import { bnOrZero } from '@shapeshiftoss/utils'
+import { partition } from 'lodash'
+import type { FC } from 'react'
+import { useMemo } from 'react'
+import type { Address } from 'viem'
+import { Text } from 'components/Text'
+import { chainIdFeeAssetReferenceMap } from 'state/slices/assetsSlice/utils'
+
+import { WithBackButton } from '../../WithBackButton'
+import { useGetLimitOrdersQuery } from '../hooks/useGetLimitOrdersForAccountQuery'
+import { LimitOrderCard } from './LimitOrderCard'
+
+type LimitOrderListProps = {
+  isLoading: boolean
+  cardProps?: CardProps
+  onBack?: () => void
+}
+
+const cowSwapTokenToAssetId = (chainId: ChainId, cowSwapToken: Address) => {
+  const { chainNamespace, chainReference } = fromChainId(chainId)
+  return cowSwapToken.toLowerCase() === COW_SWAP_NATIVE_ASSET_MARKER_ADDRESS.toLowerCase()
+    ? toAssetId({
+        chainId,
+        assetNamespace: 'slip44',
+        assetReference: chainIdFeeAssetReferenceMap(chainNamespace, chainReference),
+      })
+    : toAssetId({
+        chainId,
+        assetNamespace: ASSET_NAMESPACE.erc20,
+        assetReference: cowSwapToken,
+      })
+}
+
+export const LimitOrderList: FC<LimitOrderListProps> = ({ cardProps, onBack }) => {
+  const textColorBaseProps = useMemo(() => {
+    return {
+      color: 'text.base',
+      ...(onBack && {
+        bg: 'blue.500',
+        px: 4,
+        py: 2,
+        borderRadius: 'full',
+      }),
+    }
+  }, [onBack])
+
+  const { data: ordersResponse, isLoading } = useGetLimitOrdersQuery()
+
+  const [openLimitOrders, historicalLimitOrders] = useMemo(() => {
+    if (!ordersResponse) return []
+    return partition(
+      ordersResponse
+        .filter(({ order }) => {
+          // Parse appData json to extract the order class
+          const appDataOrderClass = JSON.parse(order.fullAppData ?? '{}').metadata?.orderClass
+            ?.orderClass as OrderClass
+          // Prefer appData as the source of truth for order class, falling back to `order.class`.
+          // Required because they often differ, and appData has the most correct information.
+          const orderClass = appDataOrderClass ?? order.class
+          return orderClass === OrderClass.LIMIT && order.signingScheme === SigningScheme.EIP712
+        })
+        .map(({ accountId, order }) => {
+          const { chainId } = fromAccountId(accountId)
+          const sellAssetId = cowSwapTokenToAssetId(chainId, order.sellToken)
+          const buyAssetId = cowSwapTokenToAssetId(chainId, order.buyToken)
+          return { accountId, sellAssetId, buyAssetId, order }
+        }),
+      ({ order }) => [OrderStatus.OPEN, OrderStatus.PRESIGNATURE_PENDING].includes(order.status),
```

Comment:
q: Is `PRESIGNATURE_PENDING` an open limit order?
Looks like this may only be a thing for SAFE signers (which would need its own PR to handle) *and* is not actually an opened order just yet at this stage but I could be wrong here, not really documented.
With that being said, probably doesn't hurt to keep it either way for the time being.

https://github.com/cowprotocol/cowswap/blob/91bab8dc3b124a32790c3bb1955b01dd3a75ea7b/apps/cowswap-frontend/src/common/pure/TransactionSubmittedContent/index.tsx#L70